xoa/data_source_vms.go

```
@@ -59,10 +58,16 @@ func dataSourceVmsRead(d *schema.ResourceData, m interface{}) error {
 
 }
 
-func vmToMapList(vms []client.Vm) []map[string]interface{} {
+func vmToMapList(vms []client.Vm, c client.XOClient) []map[string]interface{} {
 	result := make([]map[string]interface{}, 0, len(vms))
 	for _, vm := range vms {
 		log.Printf("[DEBUG] IPS %s\n", vm.Addresses)
+		disk := disksToMapList(vm.Disks)
+		var network []map[string]interface{}
+		vifs, err := c.GetVIFs(&vm)
+		if err == nil {
+			network = vifsToMapList(vifs, extractIpsFromNetworks(vm.Addresses))
+		}
```

I think error handling is best done outside of these `*toMapList` functions. The downside to fetching the data inside this transformation function is that errors are swallowed.

We could do something like the following so errors can be returned: change the `vmToMapList` function signature and handle errors inside the `dataSourceVmsRead` function.

```
func vmToMapList(vms []client.Vm, vmsVifs []client.Vif, disks []client.Disk, ) []map[string]interface{} {
 [...]
 result := make([]map[string]interface{}, 0, len(vms))
 for i, vm := range vms {
   disk := disksToMapList(vm.Disks)
   network := vifsToMapList(vmsVifs[i], extractIpsFromNetworks(vm.Addresses))
  }
}

func dataSourceVmsRead(d *schema.ResourceData, m interface{}) error { 
 [...]
 vmsVifs := [][]client.VIF{}
 for _, vm := vms {
   vifs, err := c.GetVIFs(&vm)
   if err != nil {
     return err
   }
   vmsVifs = append(vifs, v)
 }

 if err = d.Set("vms", vmToMapList(vms, vmsVifs)); err != nil {
    return err
 }
 
}
```